influx-reports/influx.py

```diff
@@ -0,0 +1,166 @@
+
+    client = InfluxDBClient(influx_ip, influx_port, database='telegraf')
+
+    # First find the first and last timestamp from telegraf.
+    # This way querying maas_* data will always
+    # be accurate (if no data is reported by maas plugins,
+    # this should be a failure metric)
+    find_first_timestamp_query = ("select first(total) from processes "
+                                  "where job_reference = '{}';".format(job_reference))
+    first_ts = return_time(client, find_first_timestamp_query, delta_seconds=5)
+    find_last_timestamp_query = ("select last(total) from processes "
+                                 "where job_reference = '{}';".format(job_reference))
+    last_ts = return_time(client, find_last_timestamp_query, delta_seconds=-5)
+
+    logging.info(
+        "Metrics were gathered between {} and {}".format(first_ts, last_ts))
+
+    measurements = ['maas_glance', 'maas_cinder', 'maas_keystone', 'maas_heat', 'maas_neutron',
+                    'maas_nova', 'maas_horizon']
+
+    yml_report = False
+    subunit_report = False
+
+    if args.ymlreport:
+        yml_report = True
+        output_file = open(args.ymlreport, 'w+')
+        output_file.write("---\n")
+    if args.subunitreport:
+        subunit_report = True
+        output_stream = open(args.subunitreport, 'wb')
+        output = StreamResultToBytes(output_stream)
+        output.startTestRun()
+
+    for measurement in measurements:
+        measurement_data = return_uptime(client, measurement, job_reference)
+        yml_output = yaml.safe_dump(
+            {measurement: dict(measurement_data)}, default_flow_style=False)
+        logging.info(yml_output)
+        if yml_report:
+            output_file.write(yml_output)
+        if subunit_report:
+            for measure in measurement_data:
+                # Input status
+                output.status(
+                    test_id=measure,
+                    timestamp=first_ts
+                )
+
+                # Output the end of the event
+                output.status(
+                    test_id=measure,
+                    # TODO(evrardjp): Define a Threshold to compare with
+                    # and use it as a success criteria.
+                    test_status="success",
+                    test_tags=None,
+                    runnable=False,
+                    # file_name=measure,
+                    # file_bytes=str(measurement_data[measure]).encode('utf-8'),
+                    timestamp=last_ts,
+                    eof=True,
+                    mime_type='text/plain; charset=UTF8'
+                )
+    if subunit_report:
+        output.stopTestRun()
+        output_stream.close()
+    if yml_report:
+        output_file.close()
+
+
+if __name__ == "__main__":
+    """ Args parser and logic router """
+    logging.getLogger().setLevel(logging.INFO)
+    parser = argparse.ArgumentParser(
+        description='Fetch maas_ metrics, and report downtime data.')
```

Comment:
If these numbers represent uptime in seconds, it would be useful to know what the maximum possible uptime was or the percentage of uptime. Without the knowing the period of time measured it's difficult to know whether or not there is an issue.